src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.Windows.cs

```diff
@@ -131,14 +132,52 @@ public NamedPipeTest_CurrentUserOnly_Windows(TestAccountImpersonator testAccount
             _testAccountImpersonator = testAccountImpersonator;
         }
 
+        [OuterLoop]
+        [ConditionalFact(nameof(IsAdminOnSupportedWindowsVersions))]
+        public async Task Connection_UnderDifferentUsers_CurrentUserOnlyOnServer_InvalidClientConnectionAttempts_DoNotBlockSuccessfulClient()
+        {
+            string name = PipeStreamConformanceTests.GetUniquePipeName();
+            bool invalidClientShouldStop = false;
+            bool invalidClientIsRunning = false;
+
+            using var server = new NamedPipeServerStream(name, PipeDirection.InOut, 1, PipeTransmissionMode.Byte, PipeOptions.Asynchronous | PipeOptions.CurrentUserOnly);
+            Task serverWait = server.WaitForConnectionAsync();
+
+            Task invalidClient = Task.Run(() =>
+            {
+                // invalid non-current user tries to connect to server.
+                _testAccountImpersonator.RunImpersonated(() =>
+                {
+                    while (!Volatile.Read(ref invalidClientShouldStop))
+                    {
+                        using var client = new NamedPipeClientStream(".", name, PipeDirection.In, PipeOptions.Asynchronous);
+                        Assert.Throws<UnauthorizedAccessException>(() => client.Connect());
+                        Volatile.Write(ref invalidClientIsRunning, true);
+                    }
+                });
+            });
+
+            Assert.False(serverWait.IsCompleted);
+            while (!Volatile.Read(ref invalidClientIsRunning)) ; // Wait until the invalid client starts running.
+
+            // valid client tries to connect and succeeds.
+            using var client = new NamedPipeClientStream(".", name, PipeDirection.In, PipeOptions.Asynchronous);
+            client.Connect();
```

Comment:
Is performing the blocking synchronous `Connect` call done on purpose here?

```suggestion
            await client.ConnectAsync();
```

Reply:
Yes, my intent was that `server.WaitForConnectionAsync()` is on thread 1, the invalid client is on thread 2, and valid client on main thread.